Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -138,4 +140,39 @@ AnimationTimeline::ShouldUpdateAnimationsAndSendEvents ScrollTimeline::documentW
     return AnimationTimeline::ShouldUpdateAnimationsAndSendEvents::No;
 }
 
+ScrollableArea* ScrollTimeline::scrollableAreaForSourceRenderer(CheckedPtr<RenderBox> renderer, Ref<Document> document)
+{
+    return renderer->element() == document->documentElement() ? (ScrollableArea*)&renderer->view().frameView() : (ScrollableArea*)renderer->layer()->scrollableArea();
+}
+
+ScrollTimeline::Data ScrollTimeline::computeScrollTimelineData() const
+{
+    if (!m_source)
+        return { };
+
+    CheckedPtr renderBox = dynamicDowncast<RenderBox>(m_source->renderer());
+    if (!renderBox)
+        return { };
+
+    auto* sourceScrollableArea = scrollableAreaForSourceRenderer(renderBox, m_source->document());
+    if (!sourceScrollableArea)
+        return { };
+
+    // https://drafts.csswg.org/scroll-animations-1/#scroll-timeline-progress
+    // Progress (the current time) for a scroll progress timeline is calculated as:
+    // scroll offset ÷ (scrollable overflow size − scroll container size)
+
+    float scrollContainerSize = axis() == ScrollAxis::Block ? sourceScrollableArea->visibleHeight() : sourceScrollableArea->visibleWidth();
+    float scrollOffset = axis() == ScrollAxis::Block ? sourceScrollableArea->scrollPosition().y() : sourceScrollableArea->scrollPosition().x();
+    float scrollableOverflowSize = m_axis == ScrollAxis::Block ? renderBox->scrollHeight() : renderBox->scrollWidth();
+
+    return { scrollableOverflowSize , scrollContainerSize, scrollOffset };
+}
+
+std::optional<Seconds> ScrollTimeline::currentTime()
+{
+    auto data = computeScrollTimelineData();
+    return Seconds { data.scrollOffset / (data.scrollableOverflowSize - data.scrollContainerSize) };
```

Comment:
`data.scrollableOverflowSize - data.scrollContainerSize` is exactly `ScrollableArea::maximumScrollOffset()` so you should use that from the get-go.

Reply:
Fixed.